### DISCUSSION TITLE
advantages to this crate over slqx or diesel?

### DISCUSSION THREAD

**Discussion by evbo** [blackbeam/rust-mysql-simple#361] [General]
Newcomer to this repo and just thought I'd ask some basic marketing questions:

- For simple use cases where I want a minimal code footprint (AWS Lambda inserting a few records into a MySQL database) does this crate likely have a lower code footprint than say the more feature-ful crates such as sqlx or diesel?
- I've heard tell sqlx can be a little slow relative to diesel and I'm wondering where this crate may land on the spectrum? Or is it more suited for certain MySQL operations but likely slower at others?
- In general, why would someone pick this crate over sqlx or diesel? No need to mention the name, "...simple", as clearly it already implies there are fewer moving parts, which is always nice in consideration of lowering attack surface and bugs.

These are just questions to help me settle in here before I learn by doing. Also, hopefully others may find them useful when deciding which crate is best for their use case.

**Comment by blackbeam** [comment]
Hi.

> For simple use cases where I want a minimal code footprint (AWS Lambda inserting a few records into a MySQL database) does this crate likely have a lower code footprint than say the more feature-ful crates such as sqlx or diesel?

Hard to say for sure. Please look at the list of `mysql_common` features and disable everything unrelated to your use case.

> I've heard tell sqlx can be a little slow relative to diesel and I'm wondering where this crate may land on the spectrum? Or is it more suited for certain MySQL operations but likely slower at others?

Diesel contributors made some benchmarks - https://github.com/diesel-rs/metrics. You need to look at `mysql_by_id` and `mysql_by_name` or just `mysql` for some sections. Is suggest you too look into the source code for the actual meaning of these benchmarks. `diesel-...` lines, I believe, should roughly represent libmysqlclient performance.

> In general, why would someone pick this crate over sqlx or diesel? No need to mention the name, "...simple", as clearly it already implies there are fewer moving parts, which is always nice in consideration of lowering attack surface and bugs.

Well, `mysql` crate is a general purpose MySQL driver i.e. you should be able implement something like `diesel` on top of it. There are some convenience additions though like FromRow trait and macros. Anyway, answer depends one the someone's requirements. The list of features is [here](https://docs.rs/mysql/latest/mysql/) (note that binlog is also supported but not listed).
